Source/WebCore/html/HTMLVideoElement.cpp

```diff
@@ -345,6 +346,14 @@ RefPtr<ImageBuffer> HTMLVideoElement::createBufferForPainting(const FloatSize& s
     return ImageBuffer::create(size, RenderingPurpose::MediaPainting, 1, colorSpace, pixelFormat, bufferOptionsForRendingMode(renderingMode), hostWindow);
 }
 
+void HTMLVideoElement::paint(GraphicsContext& context, const FloatRect& destRect)
+{
+    RefPtr player = this->player();
+    if (!player)
+        return;
+    player->playerPrivate()->paint(context, destRect);
```

Comment:
```suggestion
    player->paint(context, destRect);
```

Reply:
Also, if we do this, we probably end up in MediaPlayerPrivateRemote::paint which triggers two sync IPC, one to get the video frame, and another one to convert it to native video frame?
Is there a way to remove the conversion and call context. drawVideoFrame instead of context.drawNativeImage?

Reply:
> Is there a way to remove the conversion and call context. drawVideoFrame instead of context.drawNativeImage?

Yes, good catch. That was the point of the patch all along, I just assumed it was already working like this but of it wasn't.  Updated..